resources/codegen/tests/test___main__.py

```diff
@@ -0,0 +1,202 @@
+        assert execute_cli("-o", tmp_path, yaml_path_all).returncode == 0
+
+    def test_when_invalid_input_path_then_exit_code_is_one(self, yaml_path_all, tmp_path):
+        assert execute_cli("-o", tmp_path, "invalid").returncode == 1
+
+    def test_when_invalid_args_set_then_exit_code_is_two(self):
+        assert execute_cli("--invalid-arg").returncode == 2
+
+
+class Test_CLI_output:
+    def test_when_verbose_off_then_stdout_is_empty(self, yaml_path_minimal, tmp_path):
+        assert execute_cli("--python", "-o", tmp_path, yaml_path_minimal).stdout == ""
+
+    def test_when_verbose_on_then_created_paths_written_to_stdout(
+        self, yaml_path_minimal, creations_minimal_python, tmp_path
+    ):
+        expected = [os.path.join(tmp_path, path) for path in creations_minimal_python]
+        actual = execute_cli(
+            "--python", "-o", tmp_path, yaml_path_minimal, "-v"
+        ).stdout.splitlines()
+
+        assert actual == expected
+
+    def test_when_verbose_and_logging_on_then_created_paths_written_to_stdout(
+        self, yaml_path_minimal, creations_minimal_python, tmp_path
+    ):
+        expected = [os.path.join(tmp_path, path) for path in creations_minimal_python]
+        actual = execute_cli(
+            "--python", "-o", tmp_path, yaml_path_minimal, "-v", "-l", "DEBUG"
+        ).stdout.splitlines()
+
+        assert actual == expected
+
+    def test_when_logging_not_set_then_INFO_messages_not_written_to_stderr(
+        self, yaml_path_minimal, tmp_path
+    ):
+        assert execute_cli("--python", "-o", tmp_path, yaml_path_minimal).stderr.splitlines() == []
+
+    def test_when_logging_set_to_INFO_then_INFO_messages_written_to_stderr(
+        self, yaml_path_minimal, tmp_path
+    ):
+        assert (
+            "INFO:"
+            in execute_cli("--python", "-o", tmp_path, "-l", "INFO", yaml_path_minimal).stderr
+        )
+
+
+class Test_CLI_args_dry_run:
+    def test_when_not_set_then_code_is_generated_to_expected_path(
+        self, yaml_path_minimal, tmp_path
+    ):
+        execute_cli("--python", "-o", tmp_path, yaml_path_minimal)
+
+        assert os.path.isdir(os.path.join(tmp_path, "python"))
+
+    def test_when_d_set_then_code_is_not_generated(self, yaml_path_minimal, tmp_path):
+        execute_cli("--python", "-d", "-o", tmp_path, yaml_path_minimal)
+
+        assert not os.path.isdir(os.path.join(tmp_path, "python"))
+
+    def test_when_dryrun_set_then_code_is_not_generated(self, yaml_path_minimal, tmp_path):
+        execute_cli("--python", "--dry-run", "-o", tmp_path, yaml_path_minimal)
+
+        assert not os.path.isdir(os.path.join(tmp_path, "python"))
+
+    def test_when_set_then_exit_code_is_zero(self, yaml_path_minimal, tmp_path):
+        assert execute_cli("--python", "-d", "-o", tmp_path, yaml_path_minimal).returncode == 0
+
+
+class Test_CLI_args_output_dir:
+    def test_when_not_set_then_exit_code_is_two(self):
+        assert execute_cli("invalid").returncode == 2
```

Comment:
[consider] Perhaps make this command line look more like the others  (but without the `-o`), so its clearer what the reason for the failure must be.

Reply:
Good call, updated [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/529/commits/5e2f4fad94e01dee20cf691d37e8d254d0a45154).